tests/test_visitation.py

```diff
@@ -273,5 +275,33 @@ def test_job_initialize(self):
                 self.assertEquals(num_work_ids, len(set(r.work_ids)))
 
 
+@testmode.integration
+class TestIntegration(unittest.TestCase):
+
+    def test_storage_verification(self):
+        self._test('storage', '--replica=aws', '--replica=gcp', 'verify')
+
+    def test_index_verification(self):
+        self._test('index', '--replica=aws', '--prefix=42', 'verify')
+
+    def _test(self, *args):
+        command = (f'{pkg_root}/scripts/admin-cli.py',) + args
+        logger.info('Running %r', command)
+        input = subprocess.check_output(command)
+        logger.info('Step function input: %s', input)
+        input = json.loads(input.decode())
+        arn = input['arn']
+        output = self._get_execution_output(arn)
+        logger.info('Step function output: %r', output)
+        self.assertIn('work_result', output)
```

Comment:
Correct, it does not verify the logic but it would be difficult to set-up test buckets and very difficult to verify that the lambda emitted the expected logs. I would like to keep this test as is because it makes sure that all the machinery (admin CLI, admin lambda, visitation lambda, visitation code, concrete visitations) fits together.

Let me add a unit test that covers this code but uses mocks instead of real buckets.